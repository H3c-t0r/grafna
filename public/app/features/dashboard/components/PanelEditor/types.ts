// Us
export enum DisplayMode {
  Full = 0,
  Fit = 1,
  Exact = 2,
}

export const displayModes = [
  { value: DisplayMode.Full, label: 'Full', description: 'Use all avaliable space' },
  { value: DisplayMode.Fit, label: 'Fit', description: 'Fit in the space keeping ratio' },
  { value: DisplayMode.Exact, label: 'Exact', description: 'Same size as the dashboard' },
];
