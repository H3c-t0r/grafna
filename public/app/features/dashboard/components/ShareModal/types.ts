import React from 'react';

import { NavModelItem } from '@grafana/data';
import { DashboardModel, PanelModel } from 'app/features/dashboard/state';
import { DashboardGridItem } from 'app/features/dashboard-scene/scene/DashboardGridItem';
import { DashboardScene } from 'app/features/dashboard-scene/scene/DashboardScene';

export interface ShareModalTabProps {
  dashboard: DashboardModel;
  dashboardScene?: DashboardScene;
  gridItem?: DashboardGridItem;
  panel?: PanelModel;
  onDismiss?(): void;
}

export interface ShareModalTabModel {
  label: string;
  value: string;
  tabSuffix?: NavModelItem['tabSuffix'];
  component: React.ComponentType<ShareModalTabProps>;
}
