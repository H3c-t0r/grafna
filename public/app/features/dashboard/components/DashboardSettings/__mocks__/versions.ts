export const versions = [
  {
    id: 239,
    dashboardId: 74,
    parentVersion: 6,
    restoredFrom: 0,
    version: 7,
    created: '2021-01-14T15:14:25+01:00',
    createdBy: 'admin',
    message: '',
  },
  {
    id: 237,
    dashboardId: 74,
    parentVersion: 5,
    restoredFrom: 0,
    version: 6,
    created: '2021-01-14T14:55:29+01:00',
    createdBy: 'admin',
    message: '',
  },
  {
    id: 236,
    dashboardId: 74,
    parentVersion: 4,
    restoredFrom: 0,
    version: 5,
    created: '2021-01-14T14:28:01+01:00',
    createdBy: 'admin',
    message: '',
  },
  {
    id: 218,
    dashboardId: 74,
    parentVersion: 3,
    restoredFrom: 0,
    version: 4,
    created: '2021-01-08T10:45:33+01:00',
    createdBy: 'admin',
    message: '',
  },
  {
    id: 217,
    dashboardId: 74,
    parentVersion: 2,
    restoredFrom: 0,
    version: 3,
    created: '2021-01-05T15:41:33+01:00',
    createdBy: 'admin',
    message: '',
  },
  {
    id: 216,
    dashboardId: 74,
    parentVersion: 1,
    restoredFrom: 0,
    version: 2,
    created: '2021-01-05T15:01:50+01:00',
    createdBy: 'admin',
    message: '',
  },
  {
    id: 215,
    dashboardId: 74,
    parentVersion: 1,
    restoredFrom: 0,
    version: 1,
    created: '2021-01-05T14:59:15+01:00',
    createdBy: 'admin',
    message: '',
  },
];
