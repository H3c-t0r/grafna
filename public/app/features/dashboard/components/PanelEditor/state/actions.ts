import { PanelModel } from '../../../state/PanelModel';
import { PanelData } from '@grafana/data';
import { ThunkResult } from 'app/types';
import { setEditorPanelData, updateEditorInitState } from './reducers';

export function initPanelEditor(sourcePanel: PanelModel): ThunkResult<void> {
  return dispatch => {
    const panel = sourcePanel.getEditClone();

    const queryRunner = panel.getQueryRunner();
    const querySubscription = queryRunner.getData().subscribe({
      next: (data: PanelData) => dispatch(setEditorPanelData(data)),
    });

    dispatch(
      updateEditorInitState({
        panel,
        sourcePanel,
        querySubscription,
      })
    );
  };
}

export function panelEditorCleanUp(): ThunkResult<void> {
  return (dispatch, getStore) => {
    const dashboard = getStore().dashboard.getModel();
    const { getPanel, querySubscription } = getStore().panelEditorNew;

    dashboard.updatePanel(getPanel());

    querySubscription.unsubscribe();
  };
}
