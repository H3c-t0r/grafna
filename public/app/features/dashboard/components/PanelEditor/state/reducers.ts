import { Unsubscribable } from 'rxjs';
import { createSlice, PayloadAction } from '@reduxjs/toolkit';
import { PanelModel } from '../../../state/PanelModel';
import { PanelData, LoadingState, DefaultTimeRange } from '@grafana/data';
import { DisplayMode } from '../types';

export interface PanelEditorStateNew {
  getSourcePanel: () => PanelModel;
  getPanel: () => PanelModel;
  data: PanelData;
  mode: DisplayMode;
  isPanelOptionsVisible: boolean;
  querySubscription?: Unsubscribable;
  initDone: boolean;
}

export const initialState: PanelEditorStateNew = {
  getPanel: () => new PanelModel({}),
  getSourcePanel: () => new PanelModel({}),
  data: {
    state: LoadingState.NotStarted,
    series: [],
    timeRange: DefaultTimeRange,
  },
  isPanelOptionsVisible: true,
  mode: DisplayMode.Fill,
  initDone: false,
};

interface InitEditorPayload {
  panel: PanelModel;
  sourcePanel: PanelModel;
  querySubscription: Unsubscribable;
}

const pluginsSlice = createSlice({
  name: 'panelEditorNew',
  initialState,
  reducers: {
    updateEditorInitState: (state, action: PayloadAction<InitEditorPayload>) => {
      state.getPanel = () => action.payload.panel;
      state.getSourcePanel = () => action.payload.sourcePanel;
      state.querySubscription = action.payload.querySubscription;
      state.initDone = true;
    },
    setEditorPanelData: (state, action: PayloadAction<PanelData>) => {
      state.data = action.payload;
    },
  },
});

export const { updateEditorInitState, setEditorPanelData } = pluginsSlice.actions;

export const panelEditorReducerNew = pluginsSlice.reducer;
