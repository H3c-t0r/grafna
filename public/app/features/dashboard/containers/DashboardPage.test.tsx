import React from 'react';
import { Provider } from 'react-redux';
import { render, screen } from '@testing-library/react';
import { UnthemedDashboardPage, mapStateToProps, Props } from './DashboardPage';
import { Router } from 'react-router-dom';
import { locationService } from '@grafana/runtime';
import { DashboardModel } from '../state';
import { configureStore } from '../../../store/configureStore';
import { mockToolkitActionCreator } from 'test/core/redux/mocks';
import { DashboardInitPhase, DashboardRoutes } from 'app/types';
import { notifyApp } from 'app/core/actions';
import { cleanUpDashboardAndVariables } from '../state/actions';
import { selectors } from '@grafana/e2e-selectors';
import { getRouteComponentProps } from 'app/core/navigation/__mocks__/routeProps';
import { createTheme } from '@grafana/data';

jest.mock('app/features/dashboard/components/DashboardSettings/GeneralSettings', () => {
  class GeneralSettings extends React.Component<{}, {}> {
    render() {
      return <>general settings</>;
    }
  }

  return { GeneralSettings };
});

jest.mock('app/core/core', () => ({
  appEvents: {
    subscribe: () => {
      return { unsubscribe: () => {} };
    },
  },
}));

interface ScenarioContext {
  cleanUpDashboardAndVariablesMock: typeof cleanUpDashboardAndVariables;
  dashboard?: DashboardModel | null;
  setDashboardProp: (overrides?: any, metaOverrides?: any) => void;
  wrapper?: HTMLElement;
  mount: (propOverrides?: Partial<Props>) => void;
  unmount: () => void;
  rerender: (propOverrides?: Partial<Props>) => void;
  setup: (fn: () => void) => void;
}

function getTestDashboard(overrides?: any, metaOverrides?: any): DashboardModel {
  const data = Object.assign(
    {
      title: 'My dashboard',
      panels: [
        {
          id: 1,
          type: 'graph',
          title: 'My graph',
          gridPos: { x: 0, y: 0, w: 1, h: 1 },
        },
      ],
    },
    overrides
  );

  const meta = Object.assign({ canSave: true, canEdit: true }, metaOverrides);
  return new DashboardModel(data, meta);
}

function dashboardPageScenario(description: string, scenarioFn: (ctx: ScenarioContext) => void) {
  describe(description, () => {
    let setupFn: () => void;

    const ctx: ScenarioContext = {
      cleanUpDashboardAndVariablesMock: jest.fn(),
      setup: (fn) => {
        setupFn = fn;
      },
      setDashboardProp: (overrides?: any, metaOverrides?: any) => {
        ctx.dashboard = getTestDashboard(overrides, metaOverrides);
      },
      mount: (propOverrides?: Partial<Props>) => {
        const store = configureStore();
        const props: Props = {
          ...getRouteComponentProps({
            match: { params: { slug: 'my-dash', uid: '11' } } as any,
            route: { routeName: DashboardRoutes.Normal } as any,
          }),
          initPhase: DashboardInitPhase.NotStarted,
          isInitSlow: false,
          initDashboard: jest.fn(),
          notifyApp: mockToolkitActionCreator(notifyApp),
          cleanUpDashboardAndVariables: ctx.cleanUpDashboardAndVariablesMock,
          cancelVariables: jest.fn(),
          templateVarsChangedInUrl: jest.fn(),
          dashboard: null,
          theme: createTheme(),
        };

        Object.assign(props, propOverrides);

        ctx.dashboard = props.dashboard;
        const { container, rerender, unmount } = render(
          <Provider store={store}>
            <Router history={locationService.getHistory()}>
              <UnthemedDashboardPage {...props} />
            </Router>
          </Provider>
        );
        ctx.wrapper = container;
        ctx.rerender = (newProps?: Partial<Props>) => {
          Object.assign(props, newProps);
          rerender(
            <Provider store={store}>
              <Router history={locationService.getHistory()}>
                <UnthemedDashboardPage {...props} />
              </Router>
            </Provider>
          );
        };
        ctx.unmount = unmount;
      },
      rerender: () => {
        // will be replaced while mount() is called
      },
      unmount: () => {},
    };

    beforeEach(() => {
      setupFn();
    });

    scenarioFn(ctx);
  });
}

describe('DashboardPage', () => {
  dashboardPageScenario('Given initial state', (ctx) => {
    ctx.setup(() => {
      ctx.mount();
    });

    it('Should render nothing', () => {
      expect(ctx.wrapper).toMatchSnapshot();
    });
  });

  dashboardPageScenario('Dashboard is fetching slowly', (ctx) => {
    ctx.setup(() => {
      ctx.mount();
      ctx.setDashboardProp({
        isInitSlow: true,
        initPhase: DashboardInitPhase.Fetching,
      });
      ctx.rerender({ dashboard: ctx.dashboard });
    });

    it('Should render slow init state', () => {
      expect(ctx.wrapper).toMatchSnapshot();
    });
  });

  dashboardPageScenario('Dashboard init completed ', (ctx) => {
    ctx.setup(() => {
      ctx.mount();
      ctx.setDashboardProp({
        title: 'My dashboard - Grafana',
      });
      ctx.rerender({ dashboard: ctx.dashboard });
    });

    it('Should update title', () => {
      expect(screen.getByText('My dashboard - Grafana')).toBeInTheDocument();
    });

    it('Should render dashboard grid', () => {
      expect(ctx.wrapper).toMatchSnapshot();
    });
  });

  dashboardPageScenario('When user goes back to dashboard from edit panel', (ctx) => {
    ctx.setup(() => {
      ctx.mount();
      ctx.setDashboardProp();
      ctx.setDashboardProp({ scrollTop: 100, queryParams: { editPanel: '1' } });
      ctx.setDashboardProp({
        queryParams: {},
      });
      ctx.rerender({ dashboard: ctx.dashboard });
    });

    it('Should update model state normal state', () => {
      expect(ctx.dashboard).toBeDefined();
      // @ts-ignore typescript doesn't understand that dashboard must be defined to reach the row below
      expect(ctx.dashboard.panelInEdit).toBeUndefined();
    });
  });

  dashboardPageScenario('When dashboard has editview url state', (ctx) => {
    ctx.setup(() => {
      ctx.mount({
        queryParams: { editview: 'settings' },
      });
      ctx.setDashboardProp();
      ctx.rerender({ dashboard: ctx.dashboard });
    });

    it('should render settings view', () => {
      expect(ctx.wrapper).toMatchSnapshot();
    });
  });

  dashboardPageScenario('When dashboard unmounts', (ctx) => {
    ctx.setup(() => {
      ctx.mount();
      ctx.setDashboardProp({
        panels: [{ id: 0, type: 'graph' }],
        schemaVersion: 17,
      });
      ctx.unmount();
    });

    it('Should call clean up action', () => {
      expect(ctx.cleanUpDashboardAndVariablesMock).toHaveBeenCalledTimes(1);
    });
  });

  dashboardPageScenario('Kiosk mode none', (ctx) => {
    ctx.setup(() => {
      ctx.mount({
        queryParams: {},
      });
      ctx.setDashboardProp({
        panels: [{ id: 0, type: 'graph' }],
        schemaVersion: 17,
      });
      ctx.rerender({ dashboard: ctx.dashboard });
    });

    it('should not render dashboard navigation ', () => {
      expect(screen.queryAllByLabelText(selectors.pages.Dashboard.DashNav.nav)).toHaveLength(1);
      expect(screen.queryAllByLabelText(selectors.pages.Dashboard.SubMenu.submenu)).toHaveLength(1);
    });
  });

  dashboardPageScenario('Kiosk mode tv', (ctx) => {
    ctx.setup(() => {
      ctx.mount({
        queryParams: { kiosk: 'tv' },
      });
      ctx.setDashboardProp({
        panels: [{ id: 0, type: 'graph' }],
        schemaVersion: 17,
      });
      ctx.rerender({ dashboard: ctx.dashboard });
    });

    it('should not render dashboard navigation ', () => {
      expect(screen.queryAllByLabelText(selectors.pages.Dashboard.DashNav.nav)).toHaveLength(1);
      expect(screen.queryAllByLabelText(selectors.pages.Dashboard.SubMenu.submenu)).toHaveLength(0);
    });
  });

  dashboardPageScenario('Kiosk mode full', (ctx) => {
    ctx.setup(() => {
      ctx.mount({
        queryParams: { kiosk: true },
      });
      ctx.setDashboardProp({
        panels: [{ id: 0, type: 'graph' }],
        schemaVersion: 17,
      });
      ctx.rerender({ dashboard: ctx.dashboard });
    });

    it('should not render dashboard navigation and submenu', () => {
      expect(screen.queryAllByLabelText(selectors.pages.Dashboard.DashNav.nav)).toHaveLength(0);
      expect(screen.queryAllByLabelText(selectors.pages.Dashboard.SubMenu.submenu)).toHaveLength(0);
    });
  });

  describe('mapStateToProps', () => {
    const props = mapStateToProps({
      panelEditor: {},
      dashboard: {
        getModel: () => ({} as DashboardModel),
      },
    } as any);

    expect(props.dashboard).toBeDefined();
  });
});
