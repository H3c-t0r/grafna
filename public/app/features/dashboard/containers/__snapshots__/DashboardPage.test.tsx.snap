// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashboardPage Dashboard init completed  Should render dashboard grid 1`] = `
<div>
  <div
    class="css-1mwktlb"
  >
    <div
      aria-label="Dashboard navigation"
    >
      <div
        class="page-toolbar css-1hik5m6"
      >
        <div
          class="css-14mr6ll"
        >
          <div
            aria-hidden="true"
            class="css-1vzus6i-Icon"
          />
        </div>
        <a
          class="css-dlnelv"
          href="/?search=open"
        >
          My dashboard - Grafana
        </a>
        <div
          class="css-umstnt"
        >
          <div
            class="css-rwrh9q"
          >
            <button
              class="css-1av2c2g"
            >
              <div
                class="css-1vzus6i-Icon"
              />
            </button>
          </div>
        </div>
        <div
          class="css-umstnt"
        >
          <div
            class="css-rwrh9q"
          >
            <button
              class="css-1av2c2g"
            >
              <div
                class="css-1vzus6i-Icon"
              />
            </button>
          </div>
        </div>
        <div
          class="css-1hos3qg"
        />
        <div
          class="css-p8muss"
        >
          <button
            aria-label="Page toolbar button Add panel"
            class="toolbar-button css-1bbmb0s-toolbar-button"
          >
            <div
              class="css-1vzus6i-Icon"
            />
          </button>
        </div>
        <div
          class="css-p8muss"
        >
          <button
            aria-label="Page toolbar button Save dashboard"
            class="toolbar-button css-1bbmb0s-toolbar-button"
          >
            <div
              class="css-1vzus6i-Icon"
            />
          </button>
        </div>
        <div
          class="css-p8muss"
        >
          <button
            aria-label="Page toolbar button Dashboard settings"
            class="toolbar-button css-1bbmb0s-toolbar-button"
          >
            <div
              class="css-1vzus6i-Icon"
            />
          </button>
        </div>
        <div
          class="css-p8muss"
        >
          <div
            class="css-o99k6q"
          >
            <div
              class="button-group css-1kl6l1j"
            >
              <button
                aria-label="TimePicker Open Button"
                class="toolbar-button css-1bbmb0s-toolbar-button"
              >
                <div
                  class="css-1vzus6i-Icon"
                />
                <div
                  class="css-1i2o8lj"
                >
                  <span
                    class="css-15dl6a3"
                  >
                    <span>
                      Last 6 hours
                    </span>
                    <span
                      class="css-1n2bi9n"
                    />
                  </span>
                </div>
                <div
                  class="css-1vzus6i-Icon"
                />
              </button>
              <button
                class="toolbar-button css-1bbmb0s-toolbar-button"
              >
                <div
                  class="css-1vzus6i-Icon"
                />
              </button>
            </div>
            <div
              class="button-group css-1c76w5q refresh-picker"
            >
              <button
                aria-label="RefreshPicker run button"
                class="toolbar-button css-1bbmb0s-toolbar-button"
              >
                <div
                  class="css-1vzus6i-Icon"
                />
              </button>
              <div
                class="button-group css-1viikh2"
              >
                <button
                  aria-label="RefreshPicker interval button"
                  class="toolbar-button css-1bbmb0s-toolbar-button"
                >
                  
                  <div
                    class="css-1vzus6i-Icon"
                  />
                </button>
              </div>
            </div>
          </div>
        </div>
        <div
          class="css-p8muss"
        >
          <button
            aria-label="Page toolbar button Cycle view mode"
            class="toolbar-button css-1bbmb0s-toolbar-button"
          >
            <div
              class="css-1vzus6i-Icon"
            />
          </button>
        </div>
      </div>
    </div>
    <div
      class="css-17x4n39"
    >
      <div
        class="css-12b7m5k"
        style="position: relative; overflow: hidden; width: 100%; height: auto; min-height: 100%; max-height: 100%;"
      >
        <div
          class="scrollbar-view"
          style="position: relative; overflow: scroll; margin-right: 0px; margin-bottom: 0px; min-height: calc(100% + 0px); max-height: calc(100% + 0px);"
        >
          <div
            class="css-1xodasp"
          >
            <div
              aria-label="Dashboard submenu"
            />
            <div
              style="overflow: visible; width: 100%; height: 100%;"
            >
              <div
                class="react-grid-layout"
                style="height: 30px;"
              >
                <div
                  class="react-grid-item react-resizable"
                  data-panelid="1"
                  style="top: 0px; left: 0px; width: 100%; height: 30px; position: absolute;"
                >
                  <div
                    aria-label="Panel container title My graph"
                    class="panel-container panel-container--absolute"
                  >
                    <div
                      class="panel-header grid-drag-handle"
                    >
                      <div
                        aria-label="Panel header title item My graph"
                        class="panel-title-container"
                      >
                        <div
                          class="panel-title"
                        >
                          <span
                            class="panel-title-text"
                          >
                            My graph
                          </span>
                          <div
                            class="css-1vzus6i-Icon"
                          />
                        </div>
                      </div>
                    </div>
                  </div>
                  <span
                    class="react-resizable-handle react-resizable-handle-se"
                  />
                </div>
              </div>
            </div>
            <div
              class="resize-triggers"
            >
              <div
                class="expand-trigger"
              >
                <div
                  style="width: 1px; height: 1px;"
                />
              </div>
              <div
                class="contract-trigger"
              />
            </div>
          </div>
        </div>
        <div
          class="track-horizontal"
          style="position: absolute; height: 6px; display: none;"
        >
          <div
            class="thumb-horizontal"
            style="position: relative; display: block; height: 100%;"
          />
        </div>
        <div
          class="track-vertical"
          style="position: absolute; width: 6px; display: none;"
        >
          <div
            class="thumb-vertical"
            style="position: relative; display: block; width: 100%;"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`DashboardPage Dashboard is fetching slowly Should render slow init state 1`] = `
<div>
  <div
    class="css-1mwktlb"
  >
    <div
      aria-label="Dashboard navigation"
    >
      <div
        class="page-toolbar css-1hik5m6"
      >
        <div
          class="css-14mr6ll"
        >
          <div
            aria-hidden="true"
            class="css-1vzus6i-Icon"
          />
        </div>
        <a
          class="css-dlnelv"
          href="/?search=open"
        >
          My dashboard
        </a>
        <div
          class="css-umstnt"
        >
          <div
            class="css-rwrh9q"
          >
            <button
              class="css-1av2c2g"
            >
              <div
                class="css-1vzus6i-Icon"
              />
            </button>
          </div>
        </div>
        <div
          class="css-umstnt"
        >
          <div
            class="css-rwrh9q"
          >
            <button
              class="css-1av2c2g"
            >
              <div
                class="css-1vzus6i-Icon"
              />
            </button>
          </div>
        </div>
        <div
          class="css-1hos3qg"
        />
        <div
          class="css-p8muss"
        >
          <button
            aria-label="Page toolbar button Add panel"
            class="toolbar-button css-1bbmb0s-toolbar-button"
          >
            <div
              class="css-1vzus6i-Icon"
            />
          </button>
        </div>
        <div
          class="css-p8muss"
        >
          <button
            aria-label="Page toolbar button Save dashboard"
            class="toolbar-button css-1bbmb0s-toolbar-button"
          >
            <div
              class="css-1vzus6i-Icon"
            />
          </button>
        </div>
        <div
          class="css-p8muss"
        >
          <button
            aria-label="Page toolbar button Dashboard settings"
            class="toolbar-button css-1bbmb0s-toolbar-button"
          >
            <div
              class="css-1vzus6i-Icon"
            />
          </button>
        </div>
        <div
          class="css-p8muss"
        >
          <div
            class="css-o99k6q"
          >
            <div
              class="button-group css-1kl6l1j"
            >
              <button
                aria-label="TimePicker Open Button"
                class="toolbar-button css-1bbmb0s-toolbar-button"
              >
                <div
                  class="css-1vzus6i-Icon"
                />
                <div
                  class="css-1i2o8lj"
                >
                  <span
                    class="css-15dl6a3"
                  >
                    <span>
                      Last 6 hours
                    </span>
                    <span
                      class="css-1n2bi9n"
                    />
                  </span>
                </div>
                <div
                  class="css-1vzus6i-Icon"
                />
              </button>
              <button
                class="toolbar-button css-1bbmb0s-toolbar-button"
              >
                <div
                  class="css-1vzus6i-Icon"
                />
              </button>
            </div>
            <div
              class="button-group css-1c76w5q refresh-picker"
            >
              <button
                aria-label="RefreshPicker run button"
                class="toolbar-button css-1bbmb0s-toolbar-button"
              >
                <div
                  class="css-1vzus6i-Icon"
                />
              </button>
              <div
                class="button-group css-1viikh2"
              >
                <button
                  aria-label="RefreshPicker interval button"
                  class="toolbar-button css-1bbmb0s-toolbar-button"
                >
                  
                  <div
                    class="css-1vzus6i-Icon"
                  />
                </button>
              </div>
            </div>
          </div>
        </div>
        <div
          class="css-p8muss"
        >
          <button
            aria-label="Page toolbar button Cycle view mode"
            class="toolbar-button css-1bbmb0s-toolbar-button"
          >
            <div
              class="css-1vzus6i-Icon"
            />
          </button>
        </div>
      </div>
    </div>
    <div
      class="css-17x4n39"
    >
      <div
        class="css-12b7m5k"
        style="position: relative; overflow: hidden; width: 100%; height: auto; min-height: 100%; max-height: 100%;"
      >
        <div
          class="scrollbar-view"
          style="position: relative; overflow: scroll; margin-right: 0px; margin-bottom: 0px; min-height: calc(100% + 0px); max-height: calc(100% + 0px);"
        >
          <div
            class="css-1xodasp"
          >
            <div
              aria-label="Dashboard submenu"
            />
            <div
              style="overflow: visible; width: 100%; height: 100%;"
            >
              <div
                class="react-grid-layout"
                style="height: 30px;"
              >
                <div
                  class="react-grid-item react-resizable"
                  data-panelid="1"
                  style="top: 0px; left: 0px; width: 100%; height: 30px; position: absolute;"
                >
                  <div
                    aria-label="Panel container title My graph"
                    class="panel-container panel-container--absolute"
                  >
                    <div
                      class="panel-header grid-drag-handle"
                    >
                      <div
                        aria-label="Panel header title item My graph"
                        class="panel-title-container"
                      >
                        <div
                          class="panel-title"
                        >
                          <span
                            class="panel-title-text"
                          >
                            My graph
                          </span>
                          <div
                            class="css-1vzus6i-Icon"
                          />
                        </div>
                      </div>
                    </div>
                  </div>
                  <span
                    class="react-resizable-handle react-resizable-handle-se"
                  />
                </div>
              </div>
            </div>
            <div
              class="resize-triggers"
            >
              <div
                class="expand-trigger"
              >
                <div
                  style="width: 1px; height: 1px;"
                />
              </div>
              <div
                class="contract-trigger"
              />
            </div>
          </div>
        </div>
        <div
          class="track-horizontal"
          style="position: absolute; height: 6px; display: none;"
        >
          <div
            class="thumb-horizontal"
            style="position: relative; display: block; height: 100%;"
          />
        </div>
        <div
          class="track-vertical"
          style="position: absolute; width: 6px; display: none;"
        >
          <div
            class="thumb-vertical"
            style="position: relative; display: block; width: 100%;"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`DashboardPage Given initial state Should render nothing 1`] = `<div />`;

exports[`DashboardPage When dashboard has editview url state should render settings view 1`] = `
<div>
  <div
    class="css-1mwktlb"
  >
    <div
      aria-label="Dashboard navigation"
    >
      <div
        class="page-toolbar css-1hik5m6"
      >
        <div
          class="css-14mr6ll"
        >
          <div
            aria-hidden="true"
            class="css-1vzus6i-Icon"
          />
        </div>
        <a
          class="css-dlnelv"
          href="/?search=open"
        >
          My dashboard
        </a>
        <div
          class="css-umstnt"
        >
          <div
            class="css-rwrh9q"
          >
            <button
              class="css-1av2c2g"
            >
              <div
                class="css-1vzus6i-Icon"
              />
            </button>
          </div>
        </div>
        <div
          class="css-umstnt"
        >
          <div
            class="css-rwrh9q"
          >
            <button
              class="css-1av2c2g"
            >
              <div
                class="css-1vzus6i-Icon"
              />
            </button>
          </div>
        </div>
        <div
          class="css-1hos3qg"
        />
        <div
          class="css-p8muss"
        >
          <button
            aria-label="Page toolbar button Add panel"
            class="toolbar-button css-1bbmb0s-toolbar-button"
          >
            <div
              class="css-1vzus6i-Icon"
            />
          </button>
        </div>
        <div
          class="css-p8muss"
        >
          <button
            aria-label="Page toolbar button Save dashboard"
            class="toolbar-button css-1bbmb0s-toolbar-button"
          >
            <div
              class="css-1vzus6i-Icon"
            />
          </button>
        </div>
        <div
          class="css-p8muss"
        >
          <button
            aria-label="Page toolbar button Dashboard settings"
            class="toolbar-button css-1bbmb0s-toolbar-button"
          >
            <div
              class="css-1vzus6i-Icon"
            />
          </button>
        </div>
        <div
          class="css-p8muss"
        >
          <div
            class="css-o99k6q"
          >
            <div
              class="button-group css-1kl6l1j"
            >
              <button
                aria-label="TimePicker Open Button"
                class="toolbar-button css-1bbmb0s-toolbar-button"
              >
                <div
                  class="css-1vzus6i-Icon"
                />
                <div
                  class="css-1i2o8lj"
                >
                  <span
                    class="css-15dl6a3"
                  >
                    <span>
                      Last 6 hours
                    </span>
                    <span
                      class="css-1n2bi9n"
                    />
                  </span>
                </div>
                <div
                  class="css-1vzus6i-Icon"
                />
              </button>
              <button
                class="toolbar-button css-1bbmb0s-toolbar-button"
              >
                <div
                  class="css-1vzus6i-Icon"
                />
              </button>
            </div>
            <div
              class="button-group css-1c76w5q refresh-picker"
            >
              <button
                aria-label="RefreshPicker run button"
                class="toolbar-button css-1bbmb0s-toolbar-button"
              >
                <div
                  class="css-1vzus6i-Icon"
                />
              </button>
              <div
                class="button-group css-1viikh2"
              >
                <button
                  aria-label="RefreshPicker interval button"
                  class="toolbar-button css-1bbmb0s-toolbar-button"
                >
                  
                  <div
                    class="css-1vzus6i-Icon"
                  />
                </button>
              </div>
            </div>
          </div>
        </div>
        <div
          class="css-p8muss"
        >
          <button
            aria-label="Page toolbar button Cycle view mode"
            class="toolbar-button css-1bbmb0s-toolbar-button"
          >
            <div
              class="css-1vzus6i-Icon"
            />
          </button>
        </div>
      </div>
    </div>
    <div
      class="css-17x4n39"
    >
      <div
        class="css-12b7m5k"
        style="position: relative; overflow: hidden; width: 100%; height: auto; min-height: 100%; max-height: 100%;"
      >
        <div
          class="scrollbar-view"
          style="position: relative; overflow: scroll; margin-right: 0px; margin-bottom: 0px; min-height: calc(100% + 0px); max-height: calc(100% + 0px);"
        >
          <div
            class="css-1xodasp"
          >
            <div
              aria-label="Dashboard submenu"
            />
            <div
              style="overflow: visible; width: 100%; height: 100%;"
            >
              <div
                class="react-grid-layout"
                style="height: 30px;"
              >
                <div
                  class="react-grid-item react-resizable"
                  data-panelid="1"
                  style="top: 0px; left: 0px; width: 100%; height: 30px; position: absolute;"
                >
                  <div
                    aria-label="Panel container title My graph"
                    class="panel-container panel-container--absolute"
                  >
                    <div
                      class="panel-header grid-drag-handle"
                    >
                      <div
                        aria-label="Panel header title item My graph"
                        class="panel-title-container"
                      >
                        <div
                          class="panel-title"
                        >
                          <span
                            class="panel-title-text"
                          >
                            My graph
                          </span>
                          <div
                            class="css-1vzus6i-Icon"
                          />
                        </div>
                      </div>
                    </div>
                  </div>
                  <span
                    class="react-resizable-handle react-resizable-handle-se"
                  />
                </div>
              </div>
            </div>
            <div
              class="resize-triggers"
            >
              <div
                class="expand-trigger"
              >
                <div
                  style="width: 1px; height: 1px;"
                />
              </div>
              <div
                class="contract-trigger"
              />
            </div>
          </div>
        </div>
        <div
          class="track-horizontal"
          style="position: absolute; height: 6px; display: none;"
        >
          <div
            class="thumb-horizontal"
            style="position: relative; display: block; height: 100%;"
          />
        </div>
        <div
          class="track-vertical"
          style="position: absolute; width: 6px; display: none;"
        >
          <div
            class="thumb-vertical"
            style="position: relative; display: block; width: 100%;"
          />
        </div>
      </div>
    </div>
    <div
      class="dashboard-settings"
    >
      <div
        class="page-toolbar css-1hik5m6"
      >
        <div
          class="css-14mr6ll"
        >
          <button
            aria-label="Go Back button"
            class="css-wfwqq8"
            surface="dashboard"
          >
            <div
              class="css-1vzus6i-Icon"
            />
          </button>
        </div>
        <div
          class="css-rkdzmz"
        >
          My dashboard / Settings
        </div>
        <div
          class="css-1hos3qg"
        />
      </div>
      <div
        class="css-12b7m5k"
        style="position: relative; overflow: hidden; width: 100%; height: auto; min-height: 0; max-height: 100%;"
      >
        <div
          class="scrollbar-view"
          style="position: relative; overflow: scroll; margin-right: 0px; margin-bottom: 0px; min-height: calc(0 + 0px); max-height: calc(100% + 0px);"
        >
          <div
            class="css-18ljh8p"
          >
            <div
              class="css-17b70ns"
            >
              <aside
                class="dashboard-settings__aside"
              >
                <a
                  aria-label="Dashboard settings section item General"
                  class="dashboard-settings__nav-item active"
                >
                  <div
                    class="css-1vzus6i-Icon"
                  />
                  General
                </a>
                <a
                  aria-label="Dashboard settings section item Annotations"
                  class="dashboard-settings__nav-item"
                >
                  <div
                    class="css-1vzus6i-Icon"
                  />
                  Annotations
                </a>
                <a
                  aria-label="Dashboard settings section item Variables"
                  class="dashboard-settings__nav-item"
                >
                  <div
                    class="css-1vzus6i-Icon"
                  />
                  Variables
                </a>
                <a
                  aria-label="Dashboard settings section item Links"
                  class="dashboard-settings__nav-item"
                >
                  <div
                    class="css-1vzus6i-Icon"
                  />
                  Links
                </a>
                <a
                  aria-label="Dashboard settings section item JSON Model"
                  class="dashboard-settings__nav-item"
                >
                  <div
                    class="css-1vzus6i-Icon"
                  />
                  JSON Model
                </a>
                <div
                  class="dashboard-settings__aside-actions"
                >
                  <button
                    aria-label="Dashboard settings aside actions Save button"
                    class="css-5tn967-button"
                  >
                    <span
                      class="css-1mhnkuh"
                    >
                      Save dashboard
                    </span>
                  </button>
                </div>
              </aside>
              <div
                class="css-1hq4llk"
              >
                general settings
              </div>
            </div>
          </div>
        </div>
        <div
          class="track-horizontal"
          style="position: absolute; height: 6px; display: none;"
        >
          <div
            class="thumb-horizontal"
            style="position: relative; display: block; height: 100%;"
          />
        </div>
        <div
          class="track-vertical"
          style="position: absolute; width: 6px; display: none;"
        >
          <div
            class="thumb-vertical"
            style="position: relative; display: block; width: 100%;"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
